addons/web_editor/models/ir_attachment.py

```diff
@@ -65,6 +84,32 @@ def _compute_image_size(self):
                 attachment.image_width = 0
                 attachment.image_height = 0
 
+    def _extract_thumbnail(self):
+        for attachment in self:
+            # We can only extract thumbnails for local attachments which are
+            # videos and which don't already have a thumbnail
+            if attachment.mimetype.startswith('video/') and not attachment.thumbnail and attachment.store_fname:
+                _logger.info(
+                    'Extracting video thumbnail for attachment %s', attachment.id)
+
+                thumbnail_path = tempfile.mktemp(suffix='.jpeg')
+                extract_video_thumbnail(self._full_path(
+                    attachment.store_fname), thumbnail_path)
+
+                with open(thumbnail_path, 'rb') as f:
+                    thumbnail_data = f.read()
+
+                    thumbnail = self.create({
+                        'name': f'{attachment.id}-video-thumbnail',
+                        'mimetype': 'image/jpeg',
+                        'res_model': 'ir.attachment',
+                        'public': attachment.public,
+                        'raw': thumbnail_data,
+                    })
```

Comment:
@odoo/rd-framework-py we should offer an API to create an attachment out of a file, i.e. to os.rename/shutil.copy that file so users don't need to load the whole thing in memory